### PR TITLE
[Editorial] IOI Pyramid

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/ioi-06-pyramid.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+Consider iterating over every possible upper left corner of our pyramid's base.
+We want to find the chamber inside our pyramid's base with the smallest sum of
+heights.
+
+This boils down to doing a 2D range minimum query on every possible upper left
+corner of our chamber. The sliding window approach is as follows: when iterating
+over each row, we maintain an array of monotonic deques for
+each column that allows us to find the best chamber for each column. Then,
+we keep another monotonic deque that keeps track of the best possible result
+from our sliding window of columns.
```
Comment (SansPapyrus683) [inline solutions/gold/ioi-06-pyramid.mdx:19]: ngl explanation seems lazy

File: solutions/gold/ioi-06-pyramid.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	                          int col_2) -> int {
+		return grid_sum[row_2][col_2] - grid_sum[row_2][col_1 - 1] -
+		       grid_sum[row_1 - 1][col_2] + grid_sum[row_1 - 1][col_1 - 1];
+	};
+	/** @return sum of pyramid rectangle with upper left corner at (r, c) */
+	const auto p_sum = [&](int row, int col) -> int {
+		return rect_sum(row, col, row + a - 1, col + b - 1);
+	};
+	/** @return sum of chamber rectangle with upper left corner at (r, c) */
+	const auto c_sum = [&](int row, int col) -> int {
```
Comment (SansPapyrus683) [inline solutions/gold/ioi-06-pyramid.mdx:58]: is the `-> int` needed

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
balls

File: solutions/gold/ioi-06-pyramid.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx:19]: ok that is pretty fair, but there isn't that much to this problem other than writing the impl

File: solutions/gold/ioi-06-pyramid.mdx (hunk 2 above)
Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx:58]: i prefer having the trailing return, but i can remove it if u want

Comment (SansPapyrus683) [reply solutions/gold/ioi-06-pyramid.mdx:58]: no u right, type annotations never hurt

**Review by ryanchou-dev — COMMENTED** [review]
code looks pretty good :O

File: solutions/gold/ioi-06-pyramid.mdx [hunk 3]
```diff
@@ -0,0 +1,128 @@
+---
+id: ioi-06-pyramid
+source: IOI 2006
+title: Pyramid
+author: Justin Ji
+---
+
```
[the remark this answers is not in the document]
Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx:7]: oh shiiiii i forgot thanks for catching that 🤡

[the remark this answers is not in the document]
Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx:7]: done

File: solutions/gold/ioi-06-pyramid.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx:21]: oh yeah about this uhh do u have any tips for making the explanation more thought out, per see

cuz I was lowk blanking out on how I could do more to explain the sol

Comment (SansPapyrus683) [reply solutions/gold/ioi-06-pyramid.mdx:21]: i mean an array of monotonic deques
what is it indexed by?
what does each deque contain?
how/when do we add/pop from it?

Comment (TheGamingMousse) [reply solutions/gold/ioi-06-pyramid.mdx]: ok i went a bit more in depth with the explanation. lmk if any part of the explanation is unclear

**Review by SansPapyrus683 — APPROVED** [review]
so true bestie

**Review by ryanchou-dev — APPROVED** [review]
yea u cooked

**Pull request merged**